### PR TITLE
Array triggers write simconfig

### PR DESCRIPTION
**Pull request by GernotMaier**
Handling of new array trigger model parameters (see PR #1192 and [MR 73](https://gitlab.cta-observatory.org/cta-science/simulations/simulation-model/model_parameters/-/merge_requests/73)).

Read array_trigger parameter for each telescope type from model parameter and write the corresponding array triggers config file for sim_telarray.

- file is always written to the model path (like all other files pulled from the database) with a fixed name `array_triggers.dat`.
- explicit listing of all telescopes IDs for a given telescope type (no abbreviation as `5 to 25`)
- consciously introduced `simtel_config_writer._convert_site_parameters_to_simtel_format`, as I expect further similar extensions in future (e.g., when storing tables as astropy tables)

This PR requires MR75 to be merged and the database updated accordingly.

## Copilot summary (autogenerated)

This pull request introduces several key changes to the `simtools` project, focusing on enhancing the validation of array triggers, updating schema definitions, and adding new functionalities to the configuration writer. The most important changes include the addition of a new format checker for array trigger names, updates to the array triggers schema, and improvements to the configuration file writer to support the new schema.

### Validation Enhancements:
* [`simtools/data_model/format_checkers.py`](diffhunk://#diff-abfa9dc7adb421e96a3ae076286df886f293d97e372abce176bb80ed2624a770R29-R38): Added a new format checker `check_array_trigger_name` to validate array trigger names using a regex pattern. (`[simtools/data_model/format_checkers.pyR29-R38](diffhunk://#diff-abfa9dc7adb421e96a3ae076286df886f293d97e372abce176bb80ed2624a770R29-R38)`)
* [`simtools/utils/names.py`](diffhunk://#diff-2ac57fd2feef99d43f8b6ba52f7eec3ce0177a313cd10d5bcd2780d327909cdfR170-R186): Introduced `validate_array_element_type` to validate array element types. (`[simtools/utils/names.pyR170-R186](diffhunk://#diff-2ac57fd2feef99d43f8b6ba52f7eec3ce0177a313cd10d5bcd2780d327909cdfR170-R186)`)

### Schema Updates:
* [`simtools/schemas/model_parameters/array_triggers.schema.yml`](diffhunk://#diff-10d0c24e0810c3dbf8147d0a2fb185e67b4716c8d3de07bf563e1a667c6cb88fL8-R91): Updated the schema to include new properties for array triggers, such as `multiplicity`, `width`, `hard_stereo`, and `min_separation`. (`[simtools/schemas/model_parameters/array_triggers.schema.ymlL8-R91](diffhunk://#diff-10d0c24e0810c3dbf8147d0a2fb185e67b4716c8d3de07bf563e1a667c6cb88fL8-R91)`)

### Configuration Writer Enhancements:
* [`simtools/simtel/simtel_config_writer.py`](diffhunk://#diff-3034e81610856a9b40c66db39ddddb4740553f93ffc2e72288a01671624cf1d6L78-R82): Added support for writing array trigger definitions to a file and updated site parameters writing to include model path and telescope model. (`[[1]](diffhunk://#diff-3034e81610856a9b40c66db39ddddb4740553f93ffc2e72288a01671624cf1d6L78-R82)`, `[[2]](diffhunk://#diff-3034e81610856a9b40c66db39ddddb4740553f93ffc2e72288a01671624cf1d6L173-R179)`, `[[3]](diffhunk://#diff-3034e81610856a9b40c66db39ddddb4740553f93ffc2e72288a01671624cf1d6L271-R277)`, `[[4]](diffhunk://#diff-3034e81610856a9b40c66db39ddddb4740553f93ffc2e72288a01671624cf1d6R287-R288)`, `[[5]](diffhunk://#diff-3034e81610856a9b40c66db39ddddb4740553f93ffc2e72288a01671624cf1d6R299-R401)`)

### Testing:
* [`tests/unit_tests/data_model/test_format_checkers.py`](diffhunk://#diff-c10e3f5cebac591f500342720bc1bbcfec1391600b355243546b1da18a02faf5R24-R34): Added tests for the new `check_array_trigger_name` format checker. (`[tests/unit_tests/data_model/test_format_checkers.pyR24-R34](diffhunk://#diff-c10e3f5cebac591f500342720bc1bbcfec1391600b355243546b1da18a02faf5R24-R34)`)
* [`tests/unit_tests/simtel/test_simtel_config_writer.py`](diffhunk://#diff-9d9352064ae2316c7c595909e25a87a149d7dd3ff2899f41c08bbadfc94c94e0R83-R134): Added tests for new methods in `simtel_config_writer`, including `_get_array_trigger_for_telescope_type` and `_convert_site_parameters_to_simtel_format`. (`[tests/unit_tests/simtel/test_simtel_config_writer.pyR83-R134](diffhunk://#diff-9d9352064ae2316c7c595909e25a87a149d7dd3ff2899f41c08bbadfc94c94e0R83-R134)`)
* [`tests/unit_tests/utils/test_names.py`](diffhunk://#diff-2d729c7621e7459fbbebbca88e959b039131032520c4e104f2923c180a3287a6R102-R111): Added tests for `validate_array_element_type`. (`[tests/unit_tests/utils/test_names.pyR102-R111](diffhunk://#diff-2d729c7621e7459fbbebbca88e959b039131032520c4e104f2923c180a3287a6R102-R111)`)

### CODE REVIEW

**Comment by GernotMaier** [comment]
It is a bit annoying that all of the following applications apparently require a array_triggers.dat file:

```
FAILED tests/integration_tests/test_applications_from_config.py::test_applications_from_config[simtools-derive-mirror-rnda_psf_measurement] - AssertionError: Application failed: simtools-derive-mirror-rnda --config /tmp/pytest-of-root/pytest-14/popen-gw6/test-data0/simtools-derive-mirror-rnda-psf_measurement/tmp_config.yml
FAILED tests/integration_tests/test_applications_from_config.py::test_applications_from_config[simtools-derive-mirror-rnda_psf_no_tuning] - AssertionError: Application failed: simtools-derive-mirror-rnda --config /tmp/pytest-of-root/pytest-14/popen-gw6/test-data1/simtools-derive-mirror-rnda-psf_no_tuning/tmp_config.yml
FAILED tests/integration_tests/test_applications_from_config.py::test_applications_from_config[simtools-derive-mirror-rnda_psf_mean] - AssertionError: Application failed: simtools-derive-mirror-rnda --config /tmp/pytest-of-root/pytest-14/popen-gw8/test-data3/simtools-derive-mirror-rnda-psf_mean/tmp_config.yml
FAILED tests/integration_tests/test_applications_from_config.py::test_applications_from_config[simtools-derive-mirror-rnda_psf_random_flen] - AssertionError: Application failed: simtools-derive-mirror-rnda --config /tmp/pytest-of-root/pytest-14/popen-gw6/test-data2/simtools-derive-mirror-rnda-psf_random_flen/tmp_config.yml
FAILED tests/integration_tests/test_applications_from_config.py::test_applications_from_config[simtools-derive-psf-parameters_run] - AssertionError: Application failed: simtools-derive-psf-parameters --config /tmp/pytest-of-root/pytest-14/popen-gw6/test-data3/simtools-derive-psf-parameters-run/tmp_config.yml
FAILED tests/integration_tests/test_applications_from_config.py::test_applications_from_config[simtools-simulate-light-emission_run-layout] - AssertionError: Application failed: simtools-simulate-light-emission --config /tmp/pytest-of-root/pytest-14/popen-gw3/test-data4/simtools-simulate-light-emission-run-layout/tmp_config.yml
FAILED tests/integration_tests/test_applications_from_config.py::test_applications_from_config[simtools-validate-cumulative-psf_run] - AssertionError: Application failed: simtools-validate-cumulative-psf --config /tmp/pytest-of-root/pytest-14/popen-gw9/test-data5/simtools-validate-cumulative-psf-run/tmp_config.yml
FAILED tests/integration_tests/test_applications_from_config.py::test_applications_from_config[simtools-validate-optics_run] - AssertionError: Application failed: simtools-validate-optics --config /tmp/pytest-of-root/pytest-14/popen-gw8/test-data11/simtools-validate-optics-run/tmp_config.yml
FAILED tests/integration_tests/test_ray_tracing.py::test_ssts[SSTS-design] - simtools.runners.simtel_runner.SimtelExecutionError:
FAILED tests/integration_tests/test_ray_tracing.py::test_plot_image - simtools.runners.simtel_runner.SimtelExecutionError:
FAILED tests/integration_tests/test_applications_from_config.py::test_applications_from_config[simtools-simulate-light-emission_run-variable] - AssertionError: Application failed: simtools-simulate-light-emission --config /tmp/pytest-of-root/pytest-14/popen-gw3/test-data5/simtools-simulate-light-emission-run-variable/tmp_config.yml
FAILED tests/integration_tests/test_ray_tracing.py::test_rx - simtools.runners.simtel_runner.SimtelExecutionError:
FAILED tests/integration_tests/test_ray_tracing.py::test_integral_curve - simtools.runners.simtel_runner.SimtelExecutionError:
FAILED tests/integration_tests/test_ray_tracing.py::test_single_mirror - simtools.runners.simtel_runner.SimtelExecutionError:
```

File: simtools/simtel/simtel_config_writer.py [hunk 1]
```diff
@@ -46,6 +48,7 @@ def __init__(
         self._label = label
         self._layout_name = layout_name
         self._telescope_model_name = telescope_model_name
+        self.array_trigger_file = None
```
Comment (orelgueta) [inline simtools/simtel/simtel_config_writer.py:51]: Can you explain why this is necessary to define here instead of always writing a file on the fly?

Comment (GernotMaier) [reply simtools/simtel/simtel_config_writer.py:51]: We need to know that a array_trigger_file is written. If yes, then we use it for the telescope_model. If note, the array-trigger parameter is not set (e.g., as for optics parameters).

File: simtools/simtel/simtel_config_writer.py [hunk 2]
```diff
@@ -288,9 +299,104 @@ def _write_site_parameters(self, file, site_model):
...
+        for count, tel_name in enumerate(telescope_model.keys()):
+            telescope_type = names.get_array_element_type_from_name(tel_name)
+            trigger_per_telescope_type.setdefault(telescope_type, []).append(count + 1)
+
+        trigger_lines = {}
+        for tel_type, tel_list in trigger_per_telescope_type.items():
+            trigger_dict = self._get_array_triggers_for_telescope_type(array_triggers, tel_type)
+            trigger_lines[tel_type] = f"Trigger {trigger_dict['multiplicity']['value']} of "
+            trigger_lines[tel_type] += ", ".join(map(str, tel_list))
+            if all(trigger_dict["width"][key] is not None for key in ["value", "unit"]):
```
Comment (orelgueta) [inline simtools/simtel/simtel_config_writer.py:363]: I shortened these `if` statements a bit, but what worries me is the case in which they are `None`. We do not raise an error and instead rely on sim_telarray to fail (which will be hard to find). Should we raise an error here?

**Review by orelgueta — COMMENTED** [review]
Couple of comments, none major

File: simtools/simtel/simtel_config_writer.py (hunk 2 above)
Comment (GernotMaier) [reply simtools/simtel/simtel_config_writer.py:363]: see issue #1200

**Review by orelgueta — APPROVED** [review]
After discussions in person, approving.

**Pull request merged**